src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/OkHttpWebSocketWrapper.java

```diff
@@ -87,14 +89,29 @@ public void onMessage(WebSocket webSocket, String message) {
         @Override
         public void onClosing(WebSocket webSocket, int code, String reason) {
             onClose.invoke(code, reason);
-            closeSubject.onComplete();
+            try {
+                closeLock.lock();
+                closeSubject.onComplete();
```

Comment:
Should we check that the closeSubject hasn't already been completed with an error here?

Reply:
Calling onComplete should noop if the subject is already completed either with an error or without an error.

Reply:
@halter73 any other feedback?